src/Search/searchTools.js

```diff
@@ -0,0 +1,118 @@
+// @flow
+import uniq from "lodash/uniq";
+import flatten from "lodash/flatten";
+import type { QueryType } from "./Types/QueryType";
+
+// function which normalize input values
+export const transformString = (str: string): string => {
+  return str
+    .toLowerCase()
+    .replace(/ /g, "")
+    .normalize("NFD")
+    .replace(/[\u0300-\u036f]/g, "");
+};
+
+// function which compares data table value with searched value in terms of type and operator
+type FilterByColNameAndOp = {
+  comparedValue: string | number,
+  operator: string,
+  searchedValue: string
+};
+
+export const filterByColNameAndOp = ({
+  comparedValue,
+  operator,
+  searchedValue
+}: FilterByColNameAndOp): boolean => {
+  let res: boolean = false;
+  let formattedOp: string = operator;
+  if (formattedOp === "=") {
+    formattedOp = "==";
+  }
+  switch (typeof comparedValue) {
```

Comment:
Reorder your switch, string should be the first case because more likely to be triggered